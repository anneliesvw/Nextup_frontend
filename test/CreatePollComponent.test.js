import Vuex from 'vuex';
import { shallow, createLocalVue } from 'vue-test-utils';
import CreatePollComp from '../src/components/groups/CreatePoll.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('CreatePollComp', () => {
  let store;
  let actions;
  beforeEach(() => {
    actions = {
      addPoll: jest.fn(),
    };
    store = new Vuex.Store({
      actions,
    });
  });

  // faalt?

  it('dispatches an action when a button is clicked', () => {
    const wrapper = shallow(CreatePollComp, {
      store,
      localVue,
    });
    wrapper.setProps({ activeGroup: { groupId: 1 } });
    wrapper.find('.create-poll-btn').trigger('click');
    expect(actions.addPoll.mock.calls).toHaveLength(1);
  });
});
