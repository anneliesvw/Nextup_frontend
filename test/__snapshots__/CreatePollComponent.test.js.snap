// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePollComp should match snapshot 1`] = `
<el-dialog
  title="polls.create"
>
  <div
    class="create-event-form"
  >
    <el-form
      class="create-event"
      label-position="top"
    >
      <div
        class="create-event-input"
      >
        <el-form-item
          label="Title"
        >
          <el-input
            placeholder="Enter title here"
          />
        </el-form-item>
         
        <el-form-item
          label="Poll Options"
        >
          <div
            class="polloption-add"
          >
            <el-input
              class="polloption-name"
              placeholder="Enter option here"
            />
             
            <el-button
              type="primary"
            >
              Add option
            </el-button>
          </div>
        </el-form-item>
         
        <div
          class="option-list"
        />
         
        <el-form-item
          label="Visible for members"
        >
          <el-switch
            active-color="#ff4949"
            active-text="No"
            inactive-color="#13ce66"
            inactive-text="Yes"
          />
        </el-form-item>
         
        <el-button
          class="create-poll-btn"
          type="primary"
        >
          Create poll
        </el-button>
      </div>
    </el-form>
  </div>
</el-dialog>
`;
